lib/spack/spack/cmd/python.py

```diff
@@ -48,10 +51,47 @@ def python(parser, args, unknown_args):
     if unknown_args:
         tty.die("Unknown arguments:", " ".join(unknown_args))
 
+    # Only python has support for arguments / commands
+    has_command = args.python_args or args.python_command
+    if has_command and args.python_interpreter != "python":
+        tty.die("Use the default python interpreter for args/commands.")
+
+    # Run user choose of interpreter
+    if args.python_interpreter == "ipython":
+        return spack.cmd.python.ipython_interpreter(args)
+    return spack.cmd.python.python_interpreter(args)
+
+
+def ipython_interpreter(args):
+    """An ipython interpreter is intended to be interactive, so it doesn't
+    support running a script or arguments
+    """
+    # If the user doesn't have ipython, fallback to python
```

Comment:
I think this comment is out of date

Reply:
Removed https://github.com/spack/spack/pull/20329/commits/a3f087f912df42eb62e5236dbdadb64874099aee